src/odemis/driver/hamamatsurx.py

```diff
@@ -1124,8 +1166,7 @@ def readCommandResponse(self):
                         try:
```

Comment:
This while block is a little suspitious. Shouldn't something special happen in case of timeout? To do so, you'd need an `else`. However, the break only happens in case of exception. So either there is a specific exception which indicates the end of the message, and it's the one that should break (via an dedicated `except `), or this code almost always take 1 s, which sounds awfully long (and arbitrary).
How do you know that the response of _getReadoutCamInfo is ended?

If the only way to know is to wait long enough. Document it clearly. In particular, you could change the `timeout` name to something like `max_time`.

Reply:
Mhm yes this code gives we some headache. I was wondering whether I can calculate the amount of bytes I am waiting for, as the string length I expect should be the same. So the only reason it will fails should be after some max time, when we decide we don't want to wait any longer...and just continue

Reply:
Any suggestions here?

Reply:
If you know there are always N (5?) lines answered, you could just wait until there are N `\r` in the answer before going out of the while loop.
Ideally, the structure of the text would be slightly different at the end (eg, just `\r` instead of `\n\r`). However, the documentation is not clear about this. Do you have some (raw) logs?

Reply:
Unfortunately I don't have a raw output. I like the idea about checking the number of lines received, as this should be indeed a fixed number. Thinking about the code again, I think it is also not clear whether "rargs" might already contain some/parts of the additional info or not. I think I need to check with HW...